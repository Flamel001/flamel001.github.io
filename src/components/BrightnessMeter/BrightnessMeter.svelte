<script lang="ts">
  import Month from './Icons/Month.svelte';
  import Sun from './Icons/Sun.svelte';
  import Bar from './Components/Bar.svelte';

  export let min = 0,
    max = 1,
    optimum = 0.65,
    value = 0;
</script>

<div class="@container flex gap-7">
  <Month isDimmed={value >= optimum} />
  <Bar {min} {max} bind:value />
  <Sun isDimmed={value < optimum} />
</div>
