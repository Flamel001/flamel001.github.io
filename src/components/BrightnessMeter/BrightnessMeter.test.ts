import { expect, describe, it } from 'vitest'
import { render } from '@testing-library/svelte'
import BrightnessMeter from './BrightnessMeter.svelte'

describe('BrightnessMeter', () => {

  it('Renders with default values', () => {
    const { container } = render(BrightnessMeter)
    expect(container.innerHTML).toMatchSnapshot()
  })

  it('Left icon highlighted when value < optimum', () => {
    const { container } = render(BrightnessMeter,{value:0.5,optimum:0.6})
    expect(container.innerHTML).toMatchSnapshot()
  })

  it('Right icon is highlighted when value >= optimum', () => {
    const { container } = render(BrightnessMeter,{value:0.5,optimum:0.5})
    expect(container.innerHTML).toMatchSnapshot()
  })
})
