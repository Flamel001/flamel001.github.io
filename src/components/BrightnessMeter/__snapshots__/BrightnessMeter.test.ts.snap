// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BrightnessMeter > Left icon is highlighted when value < optimum 1`] = `
"<div><div class="@container flex gap-7 items-center"><div class="p-6"><svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M17.3087 11.1735C17.6859 10.9964 18.0882 11.3677 17.983 11.7711C17.6242 13.1447 16.919 14.4519 15.8334 15.5375C12.6094 18.7611 7.45186 18.8285 4.31137 15.6883C1.17088 12.5482 1.23945 7.38999 4.46337 4.16641C5.54905 3.08084 6.8553 2.37579 8.23012 2.01698C8.63354 1.91186 9.00382 2.31409 8.82782 2.69118C7.68271 5.13543 8.0587 8.06304 9.99808 10.0022C11.9363 11.9425 14.8642 12.3185 17.3087 11.1735Z" fill="white" fill-opacity="1"></path></svg></div><!--<Month>--> <div class="flex gap-2.5 items-center"><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[20px] w-[6px] @xs:h-[24px] @xs:w-[8px]" style="opacity: 1;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[20px] w-[6px] @xs:h-[24px] @xs:w-[8px]" style="opacity: 1;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[24px] w-[6px] @xs:h-[28px] @xs:w-[8px]" style="opacity: 1;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[36px] w-[6px] @xs:h-[42px] @xs:w-[10px]" style="opacity: 1;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[24px] w-[6px] @xs:h-[28px] @xs:w-[8px]" style="opacity: 0.6;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[20px] w-[6px] @xs:h-[24px] @xs:w-[8px]" style="opacity: 0.4;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[20px] w-[6px] @xs:h-[24px] @xs:w-[8px]" style="opacity: 0.4;"></div><!--<Line>--></div><!--<Bar>--> <div class="p-6"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none"><path fill-rule="evenodd" clip-rule="evenodd" d="M13 3C13 2.44772 12.5523 2 12 2C11.4477 2 11 2.44772 11 3V4C11 4.55228 11.4477 5 12 5C12.5523 5 13 4.55228 13 4V3ZM6.30711 4.89289C5.91658 4.50237 5.28342 4.50237 4.89289 4.89289C4.50237 5.28342 4.50237 5.91658 4.89289 6.30711L5.59289 7.00711C5.98342 7.39763 6.61658 7.39763 7.00711 7.00711C7.39763 6.61658 7.39763 5.98342 7.00711 5.59289L6.30711 4.89289ZM19.1071 6.30711C19.4976 5.91658 19.4976 5.28342 19.1071 4.89289C18.7166 4.50237 18.0834 4.50237 17.6929 4.89289L16.9929 5.59289C16.6024 5.98342 16.6024 6.61658 16.9929 7.00711C17.3834 7.39763 18.0166 7.39763 18.4071 7.00711L19.1071 6.30711ZM3 11C2.44772 11 2 11.4477 2 12C2 12.5523 2.44772 13 3 13H4C4.55228 13 5 12.5523 5 12C5 11.4477 4.55228 11 4 11H3ZM20 11C19.4477 11 19 11.4477 19 12C19 12.5523 19.4477 13 20 13H21C21.5523 13 22 12.5523 22 12C22 11.4477 21.5523 11 21 11H20ZM7.00711 18.4071C7.39763 18.0166 7.39763 17.3834 7.00711 16.9929C6.61658 16.6024 5.98342 16.6024 5.59289 16.9929L4.89289 17.6929C4.50237 18.0834 4.50237 18.7166 4.89289 19.1071C5.28342 19.4976 5.91658 19.4976 6.30711 19.1071L7.00711 18.4071ZM18.4071 16.9929C18.0166 16.6024 17.3834 16.6024 16.9929 16.9929C16.6024 17.3834 16.6024 18.0166 16.9929 18.4071L17.6929 19.1071C18.0834 19.4976 18.7166 19.4976 19.1071 19.1071C19.4976 18.7166 19.4976 18.0834 19.1071 17.6929L18.4071 16.9929ZM13 20C13 19.4477 12.5523 19 12 19C11.4477 19 11 19.4477 11 20V21C11 21.5523 11.4477 22 12 22C12.5523 22 13 21.5523 13 21V20ZM17 12C17 14.7614 14.7614 17 12 17C9.23858 17 7 14.7614 7 12C7 9.23858 9.23858 7 12 7C14.7614 7 17 9.23858 17 12Z" fill="white" fill-opacity="0.6"></path></svg></div><!--<Sun>--></div><!--<BrightnessMeter>--></div>"
`;

exports[`BrightnessMeter > Renders with default values 1`] = `
"<div><div class="@container flex gap-7 items-center"><div class="p-6"><svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M17.3087 11.1735C17.6859 10.9964 18.0882 11.3677 17.983 11.7711C17.6242 13.1447 16.919 14.4519 15.8334 15.5375C12.6094 18.7611 7.45186 18.8285 4.31137 15.6883C1.17088 12.5482 1.23945 7.38999 4.46337 4.16641C5.54905 3.08084 6.8553 2.37579 8.23012 2.01698C8.63354 1.91186 9.00382 2.31409 8.82782 2.69118C7.68271 5.13543 8.0587 8.06304 9.99808 10.0022C11.9363 11.9425 14.8642 12.3185 17.3087 11.1735Z" fill="white" fill-opacity="1"></path></svg></div><!--<Month>--> <div class="flex gap-2.5 items-center"><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[36px] w-[6px] @xs:h-[42px] @xs:w-[10px]" style="opacity: 1;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[24px] w-[6px] @xs:h-[28px] @xs:w-[8px]" style="opacity: 0.6;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[20px] w-[6px] @xs:h-[24px] @xs:w-[8px]" style="opacity: 0.4;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[20px] w-[6px] @xs:h-[24px] @xs:w-[8px]" style="opacity: 0.4;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[20px] w-[6px] @xs:h-[24px] @xs:w-[8px]" style="opacity: 0.4;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[20px] w-[6px] @xs:h-[24px] @xs:w-[8px]" style="opacity: 0.4;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[20px] w-[6px] @xs:h-[24px] @xs:w-[8px]" style="opacity: 0.4;"></div><!--<Line>--></div><!--<Bar>--> <div class="p-6"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none"><path fill-rule="evenodd" clip-rule="evenodd" d="M13 3C13 2.44772 12.5523 2 12 2C11.4477 2 11 2.44772 11 3V4C11 4.55228 11.4477 5 12 5C12.5523 5 13 4.55228 13 4V3ZM6.30711 4.89289C5.91658 4.50237 5.28342 4.50237 4.89289 4.89289C4.50237 5.28342 4.50237 5.91658 4.89289 6.30711L5.59289 7.00711C5.98342 7.39763 6.61658 7.39763 7.00711 7.00711C7.39763 6.61658 7.39763 5.98342 7.00711 5.59289L6.30711 4.89289ZM19.1071 6.30711C19.4976 5.91658 19.4976 5.28342 19.1071 4.89289C18.7166 4.50237 18.0834 4.50237 17.6929 4.89289L16.9929 5.59289C16.6024 5.98342 16.6024 6.61658 16.9929 7.00711C17.3834 7.39763 18.0166 7.39763 18.4071 7.00711L19.1071 6.30711ZM3 11C2.44772 11 2 11.4477 2 12C2 12.5523 2.44772 13 3 13H4C4.55228 13 5 12.5523 5 12C5 11.4477 4.55228 11 4 11H3ZM20 11C19.4477 11 19 11.4477 19 12C19 12.5523 19.4477 13 20 13H21C21.5523 13 22 12.5523 22 12C22 11.4477 21.5523 11 21 11H20ZM7.00711 18.4071C7.39763 18.0166 7.39763 17.3834 7.00711 16.9929C6.61658 16.6024 5.98342 16.6024 5.59289 16.9929L4.89289 17.6929C4.50237 18.0834 4.50237 18.7166 4.89289 19.1071C5.28342 19.4976 5.91658 19.4976 6.30711 19.1071L7.00711 18.4071ZM18.4071 16.9929C18.0166 16.6024 17.3834 16.6024 16.9929 16.9929C16.6024 17.3834 16.6024 18.0166 16.9929 18.4071L17.6929 19.1071C18.0834 19.4976 18.7166 19.4976 19.1071 19.1071C19.4976 18.7166 19.4976 18.0834 19.1071 17.6929L18.4071 16.9929ZM13 20C13 19.4477 12.5523 19 12 19C11.4477 19 11 19.4477 11 20V21C11 21.5523 11.4477 22 12 22C12.5523 22 13 21.5523 13 21V20ZM17 12C17 14.7614 14.7614 17 12 17C9.23858 17 7 14.7614 7 12C7 9.23858 9.23858 7 12 7C14.7614 7 17 9.23858 17 12Z" fill="white" fill-opacity="0.6"></path></svg></div><!--<Sun>--></div><!--<BrightnessMeter>--></div>"
`;

exports[`BrightnessMeter > Right icon is highlighted when value >= optimum 1`] = `
"<div><div class="@container flex gap-7 items-center"><div class="p-6"><svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M17.3087 11.1735C17.6859 10.9964 18.0882 11.3677 17.983 11.7711C17.6242 13.1447 16.919 14.4519 15.8334 15.5375C12.6094 18.7611 7.45186 18.8285 4.31137 15.6883C1.17088 12.5482 1.23945 7.38999 4.46337 4.16641C5.54905 3.08084 6.8553 2.37579 8.23012 2.01698C8.63354 1.91186 9.00382 2.31409 8.82782 2.69118C7.68271 5.13543 8.0587 8.06304 9.99808 10.0022C11.9363 11.9425 14.8642 12.3185 17.3087 11.1735Z" fill="white" fill-opacity="0.6"></path></svg></div><!--<Month>--> <div class="flex gap-2.5 items-center"><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[20px] w-[6px] @xs:h-[24px] @xs:w-[8px]" style="opacity: 1;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[20px] w-[6px] @xs:h-[24px] @xs:w-[8px]" style="opacity: 1;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[24px] w-[6px] @xs:h-[28px] @xs:w-[8px]" style="opacity: 1;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[36px] w-[6px] @xs:h-[42px] @xs:w-[10px]" style="opacity: 1;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[24px] w-[6px] @xs:h-[28px] @xs:w-[8px]" style="opacity: 0.6;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[20px] w-[6px] @xs:h-[24px] @xs:w-[8px]" style="opacity: 0.4;"></div><!--<Line>--><div class="bg-white rounded-[3px] duration-300 transition-[height,width,opacity]
  h-[20px] w-[6px] @xs:h-[24px] @xs:w-[8px]" style="opacity: 0.4;"></div><!--<Line>--></div><!--<Bar>--> <div class="p-6"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none"><path fill-rule="evenodd" clip-rule="evenodd" d="M13 3C13 2.44772 12.5523 2 12 2C11.4477 2 11 2.44772 11 3V4C11 4.55228 11.4477 5 12 5C12.5523 5 13 4.55228 13 4V3ZM6.30711 4.89289C5.91658 4.50237 5.28342 4.50237 4.89289 4.89289C4.50237 5.28342 4.50237 5.91658 4.89289 6.30711L5.59289 7.00711C5.98342 7.39763 6.61658 7.39763 7.00711 7.00711C7.39763 6.61658 7.39763 5.98342 7.00711 5.59289L6.30711 4.89289ZM19.1071 6.30711C19.4976 5.91658 19.4976 5.28342 19.1071 4.89289C18.7166 4.50237 18.0834 4.50237 17.6929 4.89289L16.9929 5.59289C16.6024 5.98342 16.6024 6.61658 16.9929 7.00711C17.3834 7.39763 18.0166 7.39763 18.4071 7.00711L19.1071 6.30711ZM3 11C2.44772 11 2 11.4477 2 12C2 12.5523 2.44772 13 3 13H4C4.55228 13 5 12.5523 5 12C5 11.4477 4.55228 11 4 11H3ZM20 11C19.4477 11 19 11.4477 19 12C19 12.5523 19.4477 13 20 13H21C21.5523 13 22 12.5523 22 12C22 11.4477 21.5523 11 21 11H20ZM7.00711 18.4071C7.39763 18.0166 7.39763 17.3834 7.00711 16.9929C6.61658 16.6024 5.98342 16.6024 5.59289 16.9929L4.89289 17.6929C4.50237 18.0834 4.50237 18.7166 4.89289 19.1071C5.28342 19.4976 5.91658 19.4976 6.30711 19.1071L7.00711 18.4071ZM18.4071 16.9929C18.0166 16.6024 17.3834 16.6024 16.9929 16.9929C16.6024 17.3834 16.6024 18.0166 16.9929 18.4071L17.6929 19.1071C18.0834 19.4976 18.7166 19.4976 19.1071 19.1071C19.4976 18.7166 19.4976 18.0834 19.1071 17.6929L18.4071 16.9929ZM13 20C13 19.4477 12.5523 19 12 19C11.4477 19 11 19.4477 11 20V21C11 21.5523 11.4477 22 12 22C12.5523 22 13 21.5523 13 21V20ZM17 12C17 14.7614 14.7614 17 12 17C9.23858 17 7 14.7614 7 12C7 9.23858 9.23858 7 12 7C14.7614 7 17 9.23858 17 12Z" fill="white" fill-opacity="1"></path></svg></div><!--<Sun>--></div><!--<BrightnessMeter>--></div>"
`;
